tests/nn/data_parallel/test_fsdp_optimizer_utils.py

```diff
@@ -51,17 +99,33 @@ def test_consolidate_optimizer(self, optim_fn, transformer):
 
         spawn_and_init(test_fn, world_sizes=[min(torch.cuda.device_count(), 4)])
 
+    @parameterized.expand(
+        [[SGD, False], [Adam, False]],
+        name_func=rename_test,
+    )
+    def test_consolidate_optimizer_diff_world_size(self, optim_fn, transformer):
+        if torch.cuda.device_count() < 4:
+            raise unittest.SkipTest("This test requires at least 4 GPUs.")
```

Comment:
why is this? Is it because each moe group must be 2 and overall PG must be 4? If so, are we testing moe group size == 1 case? In that case, will all-gather failure on world_size of 1?

Reply:
We already test moe group of size 1 which is why i did not add additional tests for it.